spec/javascripts/data_sharing_consent_spec.js

```diff
@@ -78,6 +78,27 @@ describe("Data sharing consent page", function () {
             $("#consent-policy-dropdown-bar").click();
             expect($(".consent-policy")).toBeHidden();
             expect($("#consent-policy-dropdown-icon")).toHaveClass("fa-chevron-right");
-        })
+        });
+    });
+
+    describe("analytics", function () {
+        it("should fire a \"data_consent_form_shown\" event when loading the page", function () {
+            expect(lastTrackedEvent).toBe('data_consent_form_shown');
+        });
+
+        it("should be set up to submit events when the form is submitted", function () {
+            expect(lastTrackedForm.form.attr("id")).toBe("data-sharing");
+        });
+
+        describe("when the form is submitted", function () {
```

Comment:
The catching of the form submission by the analytics framework is done by the analytics framework itself; I could build code to mimic its behavior, but it'd just end up testing fake code. Testing that we add the right things to the form submission attachment, and that the callback returns the correct results, is sufficient.